tools/fury.js

```diff
@@ -1,25 +1,73 @@
 var fs = require('fs');
 var path = require('path');
 var parser = require('../lib/fury').legacyBlueprintParser;
+var async = require('async');
+
+// TODO: Turn into a CLI argument
+var PRINT_PARSE_RESULT = false;
 
 // Process arguments
 var args = process.argv.slice(2);
 if (typeof args == 'undefined' || args.length !== 1) {
   var scriptName = path.basename(process.argv[1]);
-  console.log('usage: ' + scriptName + ' <input file>\n');
+  console.log('usage: ' + scriptName + ' <input file>|<input directory>\n');
   process.exit(0);
 }
 
-// Read & parse
-fs.readFile(args[0], 'utf8', function (err, data) {
-  if (err) throw err;
+// Creates parsing functions to be run in series
+function createParseFunction(filePath) {
+  return function(callback) {
+    var data = fs.readFileSync(filePath, 'utf8');
+    console.log('processing ' + filePath + ' ...');
+    parser.parse({ code: data }, function(error, api, warnings) {
+      if (error) {
 
-  parser.parse({ code: data }, function(error, api, warnings) {
-    if (error) {
-        console.log(JSON.stringify(error, null, 2));
-        process.exit(error.code);
-    }
+        return callback(error);
+      }
+
+      // For debug puproses
+      if (PRINT_PARSE_RESULT)
+        console.log(JSON.stringify(api, null, 2));
```

Comment:
Nit: not that there is really a styleguide for this, but the Airbnb style guide says to use curly brackets if doing a multi-line if block.

Also, would you like a heading for this in the console, like:

```javascript
console.log('Parse results:');
```